core/modules/comment/comment.module

```diff
@@ -415,9 +419,13 @@ function comment_permission() {
     ),
     'skip comment approval' => array(
       'title' => t('Skip comment approval'),
+      'restrict access' => TRUE,
+      'warning' => t('Allows to post spam or offensive comments that become immediately visible to every visitor.'),
```

Comment:
```suggestion
      'warning' => t('Allows posting spam or offensive comments that become immediately visible to every visitor.'),
```

Reply:
This description isn't accurate. 

Nobody would ever intentionally choose to authorize "spam or offensive comments", so nobody would ever check this box if we described it this way. We do often choose to authorize trusted people to post comments without approval, however, so we should do a better job of describing  what this permission actually does, and not only state the risks.

How about something more descriptive like:
`Allows comments to be posted without approval, becoming immediately visible to everyone. Because spam and offensive comments are common, this permission should only be granted to those who are trusted."

Reply:
I agree: The warning makes the permission sound like a permission to spam or post offensive comments, while the permission is just to post comments that don't need to be first approved. I would just use *Since spam and offensive comments are common, this permission should only be granted to those who are trusted, or on a site with additional spam protection measures in place.* or a shorter sentence. The warning is an addendum to the description, which should already say what the permission does.